### PR TITLE
Numerically stable loss function

### PR DESCRIPTION
**Pull request by MrNerd24**
Made the cross entropy loss numerically stable, as it was often giving nan or inf results. Now the result of the sigmoid function is multiplied by 0.99 and then 0.005 is added to it, before inputting this to cross entropy loss function.

This branch was built on top of the balancer branch, so merge balancer first.

### CODE REVIEW

File: train_model.py [hunk 1]
```diff
@@ -4,7 +4,17 @@
...
+    output = 0.99*output + 0.005
     loss = -torch.sum(target*torch.log(output) + (1-target)*torch.log(1-output))
+
+    ## For debugging nan or inf loss:
+    #lossCpu = loss.cpu().item()
+    #if math.isnan(lossCpu) or math.isinf(lossCpu):
+    #    print("Output and target that caused a nan or inf loss")
+    #    print(lossCpu)
+    #    print(output)
+    #    print(target)
```
Comment (Kasparow) [inline train_model.py]: Pitäiskö nää ottaa kuitenkin pois

**Review by Kasparow — APPROVED** [review]

File: train_model.py (hunk 1 above)
Comment (MrNerd24) [reply train_model.py]: voisihan ne...
